File: Linguagem C Exercicios/Matrizes/DIAGONAL NEGATIVOS.c
#include <stdio.h>
#include <stdlib.h>

int main()
{

    int n,cont;

    printf("Qual a ordem da matriz? ");
    scanf("%d",&n);

    int mat[n][n];

    for (int i=0;i<n;i++){
        for(int j=0;j<n;j++){
            printf("ELEMENTO [%d,%d]: ",i,j);
            scanf("%d",&mat[i][j]);
        };
    };

    printf("DIAGONAL PRINCIPAL\n");

    for(int i=0;i<n;i++){
        printf("%d ",mat[i][i]);
    };

    printf("\n");

    cont =0;
    for(int i=0;i<n;i++){
        for(int j=0;j<n;j++){
            if (mat[i][j] < 0){
                cont++;
            };
        };
    };

    printf("QUANTIDADE DE NEGATIVOS: %d",cont);


    return 0;
}

File: Linguagem Python/Matrizes/DiagonalNegativos.py
n = int(input("Qual a ordem da matriz? "))

mat: [[int]] = [[0 for x in range(n)] for x in range(n)]

for i in range (0,n):
    for j in range (0,n):
        mat[i][j] = int(input(f"Elemento [{i},{j}]: "))

print("DIAGONAL PRINCIPAL:")
for i in range(0,n):
    for j in range(0,n):
        if j==i:
            print(f"{mat[i][j]} ",end="")

print()
cont = 0;
for i in range(0,n):
    for j in range (0,n):
        if mat[i][j] < 0:
            cont = cont + 1

print(f"QUANTIDADE DE NEGATIVOS = {cont}")